clean-abap/CleanABAP.md

```diff
@@ -1261,6 +1262,24 @@ especially if you embed multiple variables in a text.
 DATA(message) = `Received an unexpected HTTP ` && status_code && ` with message ` && text.
 ```
 
+### Use && to split long text
+
+> [Clean ABAP](#clean-abap) > [Content](#content) > [Strings](#strings) > [This section](#use--to-split-long-text)
+
+```ABAP
+DATA(message) = |This is a really long text having { variable } and characters | && 
+                |exceeding the preferred limit of { character_limit } with message { text }|.
+```
+
+If the text is long and exceeds the preferred limit of 120 characters, the text should be split 
+into multiple lines. However, pipe operator does not allow splitting texts in multiple lines.
+In such scenario, && can be used. Ideally, keep your texts short and crisp avoiding such occurence.
```

Comment:
```suggestion
If a text definition is long and exceeds the preferred limit of 120 characters it should be split into multiple lines. Since ABAP does not support multi-line literals use an appropriate operator to achieve this. Depending on your case either the literal operator `&` or the concatenation operator `&&` is more appropriate.

> _Note that `&` and `&&` may differ in performance depending on whether static literals or string templates are used._

Be aware that long text in source code can be a code smell: Especially user-facing messages should be kept separate in [text repositories](https://help.sap.com/doc/abapdocu_latest_index_htm/latest/en-US/index.htm?file=abenabap_texts.htm).
```
This is a suggested change to address my comment in #213 .

Reply:
I find `Depending on your case` to be too vague for the styleguide. If there are cases in which either `&` or `&&` are preferred over each other, then they should be described here. The note implies that you should use `&` with literals and `&&` for string templates, is that all or is there more to it?

Reply:
I did not think the full technical background should be part of the guide as it is explained in the respective sections of the [documentation](https://help.sap.com/doc/abapdocu_latest_index_htm/latest/en-US/index.htm?file=abenstring_operators.htm) where all limitations and differences between `&&` and `&` can be studied.